lib/dialects/abstract/query-generator.js

```diff
@@ -1386,16 +1386,34 @@ const QueryGenerator = {
         }
         mainQueryItems.push(this.selectFromTableFragment(options, model, mainAttributes, '('+
           options.groupedLimit.values.map(value => {
-            const where = _.assign({}, options.where);
-            where[options.groupedLimit.on] = value;
+            let where = options.where || {}
+              , include;
+
+            if (Array.isArray(options.groupedLimit.on)) {
+              include = Model._validateIncludedElements({
+                include: [{
+                  attributes: [],
+                  association: options.groupedLimit.on[0],
+                  where: {
+                    [options.groupedLimit.on[1]]: value
+                  }
+                }],
+                model
+              }).include;
+            } else {
+              where = _.assign(options.where);
+              where[options.groupedLimit.on] = value;
+            }
```

Comment:
But there isn't really a main query with groupedLimit, or is it because how we handle it so it becomes a UNION with an include and then an outer include?

No, not sure it's prettier either, maybe have `on` be `{association, field}` rather than an array.